### ISSUE TITLE
ci : add Apple silicon (M1) macOS runners

### ISSUE THREAD

**Issue by ggerganov** [ggerganov/llama.cpp#3469]
We should start running the CI on these runners too:

https://github.blog/changelog/2023-10-02-github-actions-apple-silicon-m1-macos-runners-are-now-available-in-public-beta/

Example from the `ggml` repo: https://github.com/ggerganov/ggml/pull/514

Need to do it for `llama.cpp` and `whisper.cpp`

**Comment by bobqianic** [comment]
Turns out, these M1 macOS runners aren't free at all. In fact, they're pretty [pricey](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates). They cost $0.16 per minute. That's just not reasonable.

**Comment by ggerganov** [comment]
Yup, I didn't notice this is only for enterprise

![image](https://github.com/ggerganov/llama.cpp/assets/1991296/28e5b6ff-2e24-499c-8f8b-d278eb516cb2)

**Comment by ggerganov** [comment]
I guess this is back on the table:

https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/

**Comment by jeremyelalouf** [comment]
Hi @ggerganov,

Maybe I could bring my help to this issue 😃 ?

**Comment by japj** [comment]
Is this issue still relevant after #6182 was merged?

**Comment by ggerganov** [comment]
With the addition of the `ggml-org / ggml-100-mac-m4` node to `ggml-ci` this is no longer needed.
